### ISSUE TITLE
NDM pods experienced `panic: runtime error: invalid memory address or nil pointer dereference`

### ISSUE THREAD

**Issue by dargasudarshan** [openebs/node-disk-manager#178]
**What happened**:
- On sparse disks deployed applications.
- Observed NDM pods experienced `panic: runtime error: invalid memory address or nil pointer dereference`

**Anything else we need to know?**:
```
kubectl get nodes
NAME                  STATUS    ROLES     AGE       VERSION
master-goofy-clarke   Ready     master    1h        v1.10.0
node-goofy-clarke01   Ready     <none>    1h        v1.10.0
node-goofy-clarke02   Ready     <none>    1h        v1.10.0
node-goofy-clarke03   Ready     <none>    1h        v1.10.0
```

```
sudarshan_darga@cloudshell:~ (strong-eon-153112)$ kubectl get disks
NAME                                      AGE
sparse-40c88d3b5e4b3d8bf74f3d5619e0d73a   40m
sparse-85263204bf4aaa80c2808854fefd79b5   40m
sparse-d742cc676b31571d0c7983eae5d02797   40m
sudarshan_darga@cloudshell:~ (strong-eon-153112)$ kubectl get pods -n openebs
NAME                                                              READY     STATUS    RESTARTS   AGE
cstor-sparse-pool-4ffn-555755875f-fskkm                           2/2       Running   0          48m
cstor-sparse-pool-4mdy-566f5c8749-4mmb9                           2/2       Running   0          49m
cstor-sparse-pool-yhb2-7656f7856-7w5x8                            2/2       Running   0          48m
maya-apiserver-79b965cdc4-58bpt                                   1/1       Running   0          49m
openebs-ndm-lfch7                                                 1/1       Running   2          49m
openebs-ndm-sc8ct                                                 1/1       Running   0          49m
openebs-ndm-xgmr5                                                 1/1       Running   3          49m
openebs-provisioner-67c4cbd8c6-db8s9                              1/1       Running   0          49m
openebs-snapshot-operator-644c567cf6-5bfrp                        2/2       Running   0          49m
pvc-626ce19c-f2c8-11e8-9bfe-0cc47ab586ba-target-64f9949c788j8th   3/3       Running   2          34m
pvc-847d8e79-f2c7-11e8-9bfe-0cc47ab586ba-target-8468457dc8btl9k   3/3       Running   1          41m
pvc-936b779d-f2c8-11e8-9bfe-0cc47ab586ba-target-99d97b94-j25v6    3/3       Running   1          33m
pvc-9823213b-f2c7-11e8-9bfe-0cc47ab586ba-target-7669445b6bbkggg   3/3       Running   0          40m
pvc-9f61abac-f2cb-11e8-9bfe-0cc47ab586ba-target-668c66ccbcchkl4   3/3       Running   1          11m
pvc-a2a068d0-f2cb-11e8-9bfe-0cc47ab586ba-target-8684b576956znqh   3/3       Running   0          11m
pvc-b446622a-f2c8-11e8-9bfe-0cc47ab586ba-target-868f67b767bpvch   3/3       Running   1          32m
pvc-f1dcb6b9-f2c6-11e8-9bfe-0cc47ab586ba-target-6589d56d4blgv5t   3/3       Running   0          45m
pvc-f2569b5f-f2c7-11e8-9bfe-0cc47ab586ba-target-777fb599d-smsjk   3/3       Running   0          37m
pvc-fbbd51e6-f2c7-11e8-9bfe-0cc47ab586ba-target-7f4bb8fcc95hpdl   3/3       Running   0          37m
```
Panic snipped:

```
I1128 04:36:00.184813       1 event.go:44] processing new event for disk-78568db399a7dee278f0e776b3624eec action type add
I1128 04:36:00.191634       1 eventhandler.go:49] processing details for disk-78568db399a7dee278f0e776b3624eec
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xf63177]

goroutine 21 [running]:
github.com/openebs/node-disk-manager/cmd/probe.(*udevProbe).FillDiskDetails(0xc4203b6ae0, 0xc420092e00)
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/probe/udevprobe.go:164 +0x57
github.com/openebs/node-disk-manager/cmd/controller.(*Probe).FillDiskDetails(0xc42023cc30, 0xc420092e00)
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/controller/probe.go:47 +0x3e
github.com/openebs/node-disk-manager/cmd/controller.(*Controller).FillDiskDetails(0xc42033b8f0, 0xc420092e00)
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/controller/probe.go:106 +0xdf
github.com/openebs/node-disk-manager/cmd/probe.(*ProbeEvent).addDiskEvent(0xc420178288, 0xc4203e496d, 0x3, 0xc42031c120, 0x1, 0x1)
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/probe/eventhandler.go:50 +0x169
github.com/openebs/node-disk-manager/cmd/probe.(*udevProbe).listen(0xc4203b6ae0)
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/probe/udevprobe.go:191 +0x200
created by github.com/openebs/node-disk-manager/cmd/probe.(*udevProbe).Start
	/home/travis/gopath/src/github.com/openebs/node-disk-manager/cmd/probe/udevprobe.go:115 +0x3f
```

[ndm-crash.log](https://github.com/openebs/node-disk-manager/files/2623055/ndm-crash.log)

**Comment by satbirchhikara** [comment]
we allocating memory and using it without checking if memory allocation failed or not. In this case, we may end-up with NULL pointer if 1) System is under memory pressure & allocation failed or 2) system call to get device struct failed(this may failed with many reasons.  From given logs, I can not connect which one (out of above two) is happening. Nonetheless, I would like to go ahead and fix this issue (by checking if memory allocation succeeded or not)......not only this, I will go over other probes too to check if we are following standard coding practices

<Code Snippet>
func (up *udevProbe) FillDiskDetails(d *controller.DiskInfo) {
        udevDevice := newUdevProbeForFillDiskDetails(d.ProbeIdentifiers.UdevIdentifier) <<<<udevDevice may remain NULL if allocation fail.
        udevDiskDetails := udevDevice.udevDevice.DiskInfoFromLibudev()
</Code Snippet>

**Comment by akhilerm** [comment]
The issue happened due to seachest library corrupting the memory stack. It has been fixed in latest seachest release `Release-19.06.02`. The latest library has been included in build in  #287
